### ISSUE TITLE
Training on Multiple GPUs

### ISSUE THREAD

**Issue by roshan2024nar** [huggingface/instruction-tuned-sd#19]
Hello,

I have multiple gpus in my system and want to use only one or specific one how can i do that below is command im using 

accelerate launch --mixed_precision="fp16" finetune_instruct_pix2pix.py \
  --pretrained_model_name_or_path=$MODEL_ID \
  --dataset_name=$DATASET_ID \
  --use_ema \
  --resolution=512 --random_flip \
  --train_batch_size=2 --gradient_accumulation_steps=4 --gradient_checkpointing \
  --max_train_steps=150 \
  --checkpointing_steps=50 --checkpoints_total_limit=1 \
  --learning_rate=5e-05 --lr_warmup_steps=0 \
  --mixed_precision=fp16 \
  --val_image_url="https://hf.co/datasets/diffusers/diffusers-images-docs/resolve/main/mountain.png" \
  --validation_prompt="Generate a cartoonized version of the natural image" \
  --seed=42 \
  --output_dir=$OUTPUT_DIR \
  --report_to=wandb \
  --push_to_hub
  
  Things i tried on my ubantu system: 

import os
os.environ["CUDA_VISIBLE_DEVICES"] = "0"

export CUDA_VISIBLE_DEVICES=0

**Comment by sayakpaul** [comment]
Refer to https://huggingface.co/docs/accelerate/index.

**Comment by roshan2024nar** [comment]
it's not working so i disconnected all other gpu but it's still not working is there any restrictions that it will not work on  ubantu?

**Comment by sayakpaul** [comment]
No. Maybe it's a setup related problem.

**Comment by roshan2024nar** [comment]
I have used the same setup to run multiple process  of PyTorch and TensorFlow but only while running the above code i get that error in my ubuntu, any idea why it's happening ?

**Comment by sayakpaul** [comment]
No.

**Comment by sayakpaul** [comment]
There's no error trace so, I cannot provide any concrete suggestion.

**Comment by roshan2024nar** [comment]
Windows : 
![image_2024_02_16T08_06_59_312Z](https://github.com/huggingface/instruction-tuned-sd/assets/77958364/6cc9514a-cf0b-42ce-b7dc-37f96f3bfc79)
![image_2024_02_16T08_06_41_469Z](https://github.com/huggingface/instruction-tuned-sd/assets/77958364/03f0ccd8-05db-4f7a-86c9-aa7466695e63)

Ubuntu:
![Screenshot from 2024-02-16 13-39-07](https://github.com/huggingface/instruction-tuned-sd/assets/77958364/168482c1-0686-4a54-94ff-3edda440083c)

**Comment by sayakpaul** [comment]
Ubuntu shows OOM error. So, I am not sure what's the exact cause here. Maybe try decreasing the resolution and using xformers.

**Comment by roshan2024nar** [comment]
i tried that as well same error with 256 
![Screenshot from 2024-02-16 13-51-15](https://github.com/huggingface/instruction-tuned-sd/assets/77958364/b1387629-0fb2-43c3-ab2e-7334b9243ba0)
i tired  on 12gb  RTX 3060 as well as on 16gb RTX 4080

**Comment by sayakpaul** [comment]
Cannot help you further here sorry. I have exhausted my list of suggestions.

**Comment by sree3333** [comment]
@sayakpaul How could your finetuning code run with 6GB graphic card on windows but not 12GB and 16GB cards on ubuntu?
I suspect there might be some issue with your code, please throughly once
